### PR TITLE
Destination bigquery: GCS staging mode: do not ack uncommitted states

### PR DESCRIPTION
**Pull request by edgao**
hack PR for a specific workspace. Will not be merged, will prerelease publish + pin a single workspace. see https://github.com/airbytehq/oncall/issues/2734

example problematic sequence:
1. stream1 emits a few records + a state. We hold this in memory.
2. stream2 emits many records + state. We commit this to bigquery. 
3. We now ack stream2's state (correctly) _and_ stream1's state (incorrectly)

This PR fixes our logic, so that we only ack stream2's state. This PR also includes a workaround for platform sending us state messages with the wrong namespace. Specifically:
* The connection sets a custom namespace in the replication settings
* Platform sets that namespace on the catalog and all record messages
* Platform does _not_ set that namespace on state messages
* Therefore we need to match record messages to state messages using only the stream name
* Fortunately, all streams in the workspace's connections have different names, and it's not setting a stream name prefix, so this is safe.

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`bb204d2b65`](https://github.com/airbytehq/airbyte/pull/29420/commits/bb204d2b6593f7ae14b21169f3fc32b3171f4cb2)) - ✅

⏲️  Total pipeline duration: 32mn35s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| Connector version increment check                                               | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_staging_state_handling/1692058153/bb204d2b6593f7ae14b21169f3fc32b3171f4cb2/destination-bigquery/1.7.7/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:bb204d2b6593f7ae14b21169f3fc32b3171f4cb2)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`2c1bebf6d4`](https://github.com/airbytehq/airbyte/pull/29420/commits/2c1bebf6d407dcd29d2424915f4a67b24c094249)) - ✅

⏲️  Total pipeline duration: 32mn40s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| Connector version increment check                                               | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_staging_state_handling/1692112795/2c1bebf6d407dcd29d2424915f4a67b24c094249/destination-bigquery/1.7.7/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:2c1bebf6d407dcd29d2424915f4a67b24c094249)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`ed1fb1ab1a`](https://github.com/airbytehq/airbyte/pull/29420/commits/ed1fb1ab1ade53e5036db865eb505e596d88ba4b)) - ✅

⏲️  Total pipeline duration: 33mn12s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| Connector version increment check                                               | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_staging_state_handling/1692133112/ed1fb1ab1ade53e5036db865eb505e596d88ba4b/destination-bigquery/1.7.7/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:ed1fb1ab1ade53e5036db865eb505e596d88ba4b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java [hunk 1]
```diff
@@ -210,7 +210,7 @@ protected void acceptTracked(final AirbyteMessage message) throws Exception {
         } else if (BufferFlushType.FLUSH_SINGLE_STREAM.equals(flushType.get())) {
           if (stateManager.supportsPerStreamFlush()) {
             // per-stream instance can handle flush of just a single stream
-            markStatesAsFlushedToDestination();
+            markStatesAsFlushedToDestination(stream);
```
Comment (evantahler) [inline airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java:213]: This is the bug, captured right here.

**Review by evantahler — APPROVED** [review]

**Comment by edgao** [comment]
prerelease publish running https://github.com/airbytehq/airbyte/actions/runs/5880878010/job/15948128624

**Comment by edgao** [comment]
no longer needed; closing

**Pull request closed without merging**